### PR TITLE
[kotlin][client] rename ApiInfrastructureResponse to ApiResponse

### PR DESCRIPTION
**Pull request by wing328**
rename ApiInfrastructureResponse to ApiResponse 

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

cc @jimschubert (2017/09) ❤️, @dr4ke616 (2018/08) @karismann (2019/03) @Zomzog (2019/04) @andrewemery (2019/10) @4brunu (2019/11) @yutaka0m (2020/03)

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractKotlinCodegen.java [hunk 1]
```diff
@@ -98,6 +98,7 @@ public AbstractKotlinCodegen() {
         // this includes hard reserved words defined by https://github.com/JetBrains/kotlin/blob/master/core/descriptors/src/org/jetbrains/kotlin/renderer/KeywordStringsGenerated.java
         // as well as keywords from https://kotlinlang.org/docs/reference/keyword-reference.html
         reservedWords = new HashSet<>(Arrays.asList(
+                "ApiResponse", // Used in the auto-genreated api client
```
Comment (4brunu) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractKotlinCodegen.java:101]: @wing328 typo in the auto-generated

File: modules/openapi-generator/src/main/resources/kotlin-client/libraries/jvm-okhttp/infrastructure/ApiResponse.kt.mustache [hunk 2]
```diff
@@ -6,7 +6,7 @@ package {{packageName}}.infrastructure
 
 {{#nonPublicApi}}internal {{/nonPublicApi}}interface Response
 
-{{#nonPublicApi}}internal {{/nonPublicApi}}abstract class ApiInfrastructureResponse<T>(val responseType: ResponseType): Response {
+{{#nonPublicApi}}internal {{/nonPublicApi}}abstract class ApiResponse<T>(val responseType: ResponseType): Response {
```
Comment (4brunu) [inline modules/openapi-generator/src/main/resources/kotlin-client/libraries/jvm-okhttp/infrastructure/ApiResponse.kt.mustache:9]: @wing328 to make the transition easier, could you please create a TypeAlias like this?
```
@Deprecated("Renamed to ApiResponse", ReplaceWith("ApiResponse"))
 {{#nonPublicApi}}internal {{/nonPublicApi}}typealias ApiInfrastructureResponse<T> = ApiResponse<T>
```

This way the users will have a warning in the IDE that will help them to migrate the code

**Review by 4brunu — COMMENTED** [review]
Looks good to me.
I just left two comments, one improvement and one typo.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractKotlinCodegen.java (hunk 1 above)
Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractKotlinCodegen.java]: fixed via https://github.com/OpenAPITools/openapi-generator/pull/11094/commits/6700974ab49247abd76c139b997cfd5c49c10858

File: modules/openapi-generator/src/main/resources/kotlin-client/libraries/jvm-okhttp/infrastructure/ApiResponse.kt.mustache (hunk 2 above)
Comment (wing328) [reply modules/openapi-generator/src/main/resources/kotlin-client/libraries/jvm-okhttp/infrastructure/ApiResponse.kt.mustache:9]: Thanks for the tips. I thought about that too. The reason I didn't do it is that the change related to ApiInfrastructureResponse has not been included in the stable release if i'm not mistaken.

Comment (4brunu) [reply modules/openapi-generator/src/main/resources/kotlin-client/libraries/jvm-okhttp/infrastructure/ApiResponse.kt.mustache:9]: I think you are right 👍

**Pull request merged**